delta/sdk/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/directives/AuthDirectivesSpec.scala

```diff
@@ -1,39 +1,90 @@
 package ch.epfl.bluebrain.nexus.delta.sdk.directives
 
-import akka.http.scaladsl.model.{HttpResponse, StatusCodes}
+import java.time.Instant
+
+import akka.http.scaladsl.server.AuthorizationFailedRejection
 import akka.http.scaladsl.model.headers.{BasicHttpCredentials, OAuth2BearerToken}
+import akka.http.scaladsl.model.{HttpResponse, StatusCodes}
 import akka.http.scaladsl.server.Directives._
 import akka.http.scaladsl.server.Route
 import akka.http.scaladsl.testkit.ScalatestRouteTest
 import akka.util.ByteString
-import ch.epfl.bluebrain.nexus.delta.sdk.Identities
-import ch.epfl.bluebrain.nexus.delta.sdk.directives.AuthDirectivesSpec._
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
 import ch.epfl.bluebrain.nexus.delta.sdk.error.IdentityError
-import ch.epfl.bluebrain.nexus.delta.sdk.model.Label
+import ch.epfl.bluebrain.nexus.delta.sdk.model.ResourceRef.Latest
+import ch.epfl.bluebrain.nexus.delta.sdk.model.acls.{Acl, AclAddress, AclCollection}
 import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Caller.Anonymous
 import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.User
 import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.TokenRejection.InvalidAccessToken
-import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.{AuthToken, Caller, TokenRejection}
-import monix.bio.IO
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.{AuthToken, Caller, Identity, TokenRejection}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.Permission
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Label, ResourceF}
+import ch.epfl.bluebrain.nexus.delta.sdk.{Acls, Identities}
+import monix.bio.{IO, UIO}
 import monix.execution.Scheduler.Implicits.global
+import org.mockito.IdiomaticMockito
+import org.scalatest.BeforeAndAfter
 import org.scalatest.concurrent.ScalaFutures
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpecLike
 
 class AuthDirectivesSpec
-    extends AuthDirectives(identities)
-    with AnyWordSpecLike
+    extends AnyWordSpecLike
     with ScalatestRouteTest
     with ScalaFutures
-    with Matchers {
+    with Matchers
+    with IdiomaticMockito
+    with BeforeAndAfter {
+
+  val user = User("alice", Label.unsafe("wonderland"))
+
+  val identities = new Identities {
+
+    override def exchange(token: AuthToken): IO[TokenRejection, Caller] = {
+      token match {
+        case AuthToken("alice") =>
+          IO.pure(
+            Caller(user, Set(user))
+          )
+        case _                  => IO.raiseError(InvalidAccessToken)
+
+      }
+    }
+  }
+
+  val acls = mock[Acls]
```

Comment:
Why not using an acl dummy?

Reply:
Acl dummy is in sdk-testkit, which depends on sdk, we can't include sdk-testkit in sdk, because there will be circular dependencies in sbt

Reply:
In AuthDirectives, we only want to fetch the acls so maybe we could just pass a function there ?
Or extract a simpler contract from Acls and pass it there so that we can create a dummy implementation just like Identities here ?

mockito is a Pandora box we should avoid to open imho

Reply:
I believe creating a separate contract for a single test is a bit of an overhead.
We should avoid mockito if we can, but I believe in this case it's actually simpler to use it since we can't use dummy implementation in here.

Reply:
We can reuse it in the different route tests.